### PR TITLE
Improvement to sql generator, added validation

### PR DESCRIPTION
**Pull request by AledWEvans**
## What does this pull request do?

adds validation for CSV files

## What is the intent behind these changes?

to reduce risk of system template data being incorrectly configured

### CODE REVIEW

**Comment by AledWEvans** [comment]
Had to fix some tests as they did not work on a FRIDAY because code does not return date for current date and day of week was Friday.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/SessionTemplateSQLGeneratorTest.kt [hunk 1]
```diff
@@ -128,6 +130,149 @@ class SessionTemplateSQLGeneratorTest() {
...
+    val exception = assertThrows(IllegalArgumentException::class.java) {
+      sessionTemplateSQLGenerator.validateSessionLocation(sessionLocationList)
+    }
+
+    // Then
+    assertThat(exception.message).isEqualTo("Location : must have at least one level one element (prison:prison1 key:key1)!")
+  }
+
+  @Test()
+  fun `Data parsed - session template levelTwo cant have to parents validated correctly`() {
```
Comment (dhirajshettymoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/SessionTemplateSQLGeneratorTest.kt:156]: sorry very minor comment - i think you mean two.

**Review by dhirajshettymoj — COMMENTED** [review]
very minor comment around spelling added. Otherwise good to go.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/SessionTemplateSQLGeneratorTest.kt (hunk 1 above)
Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/SessionTemplateSQLGeneratorTest.kt:156]: Will correct Sir ;-)

**Review by dhirajshettymoj — APPROVED** [review]

**Pull request merged**